server/models/User.coffee

```diff
@@ -600,8 +603,20 @@ UserSchema.statics.makeNew = (req) ->
   user.set 'preferredLanguage', lang if lang[...2] isnt 'en'
   user.set 'preferredLanguage', 'pt-BR' if not user.get('preferredLanguage') and /br\.codecombat\.com/.test(req.get('host'))
   user.set 'preferredLanguage', 'zh-HANS' if not user.get('preferredLanguage') and /cn\.codecombat\.com/.test(req.get('host'))
-  user.set 'lastIP', (req.headers['x-forwarded-for'] or req.connection.remoteAddress)?.split(/,? /)[0]
-  user.set 'country', req.country if req.country
+  #user.set 'lastIP', (req.headers['x-forwarded-for'] or req.connection.remoteAddress)?.split(/,? /)[0]
+  if ip = (req.headers['x-forwarded-for'] or req.connection.remoteAddress)?.split(/,? /)[0]
+    console.log(ip)
+    geo = geoip.lookup(ip)
+    if country = geo?.country
+      geo.countryName = countryList.getName(country)
+  if geo  
+    delete geo.range
+    user.set 'geo', geo
```

Comment:
Makes sense, but if I set the properties explicitly (like user.geo.country = geo.country), wouldn't it still be the same problem if geo.lookup changes the structure of data?

Reply:
Discussed offline.  We want to avoid setting extra properties not in the user schema, so that's why we should explicitly set properties that are only in the schema.

Reply:
Discussed offline, will change the code